File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = 0
    check = []
    lost.sort()
    reserve.sort()
    
    for i in lost:
        if i in reserve:
            check.append(i)

    for i in check:
        if i in lost:
            lost.remove(i)
        if i in reserve:
            reserve.remove(i)

    for i in range(len(reserve)):
        if reserve[i]-1 in lost:
            lost.remove(reserve[i]-1)
        elif reserve[i]+1 in lost:
            lost.remove(reserve[i]+1)
    answer =  n - len(lost)   
    
    
    return answer

File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;
class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int count = 0;
		int i = 0;
		int j = 0;
		int answer = 0;
		
        //이거 때문에 계속 90이였음
        Arrays.sort(lost);
        Arrays.sort(reserve);
        
		for(i = 0; i<reserve.length; i++) {
			for(j = 0; j<lost.length; j++) {
				if(reserve[i] == lost[j]) {
					reserve[i] = -3;
					lost[j] = -10;
				}
			}
		}
		for(i = 0; i<reserve.length; i++) {
			for(j = 0; j<lost.length; j++) {
                if(reserve[i]>0){
                    if((reserve[i]-1 == lost[j]) || ((reserve[i]+1 == lost[j]))){
                        reserve[i] = -1;
					    lost[j] = -10;
                    }
                }
			}
		}
		for(i = 0; i<lost.length; i++) {
			if(lost[i] != -10) {
				count++;
			}
		}
		
		answer = n - count;
		return answer;
    }
}